Firestore/Source/Local/FSTLevelDBMutationQueue.mm

```diff
@@ -89,11 +91,12 @@ @implementation FSTLevelDBMutationQueue {
   std::shared_ptr<DB> _db;
 }
 
-+ (instancetype)mutationQueueWithUser:(FSTUser *)user
++ (instancetype)mutationQueueWithUser:(const User &)user
                                    db:(std::shared_ptr<DB>)db
                            serializer:(FSTLocalSerializer *)serializer {
-  FSTAssert(![user.UID isEqual:@""], @"UserID must not be an empty string.");
-  NSString *userID = user.isUnauthenticated ? @"" : user.UID;
+  FSTAssert(!user.is_authenticated() || !user.uid().empty(),
```

Comment:
It's surprising that this change is required. Why doesn't the old way just work? As written this seems axiomatic or at least like an invariant that User itself should mind.

Also I note that the Android port doesn't make this assertion at all (in SQLiteMutationQueue). Maybe we can just drop this?